### PR TITLE
Control Plane Logging as a default in all patterns

### PR DESCRIPTION
**Pull request by 5herlocked**
*Issue #, if available:* #118

*Description of changes:* Added control plane logging through upstream changes and minor modifications to blueprints.
Included documentation changes + instructions on how to access Control Plane logs in CloudWatch.
Control Plane logging is an in-place change to EKS clusters and should not be a breaking change.

Hard dependency on https://github.com/aws-quickstart/cdk-eks-blueprints/commit/d8262234ffcb66d748e1fd2d2273a5208cbbff67 being packaged and released.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/patterns/existing-eks-observability-accelerators/existing-eks-apiserver-observability.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Objective 
 
-This pattern aims to add Observability on top of an existing EKS cluster and adds API server monitoring, with open source managed AWS services.
+This pattern aims to add Observability and Control plane logging on top of an existing EKS cluster and adds API server monitoring, with open source managed AWS services.
```
Comment (elamaran11) [inline docs/patterns/existing-eks-observability-accelerators/existing-eks-apiserver-observability.md:5]: This change is not required. I would rather add a bullet to say. If Control Plane logging is not enabled.. please use enableControlPlane logging function attaching a small relevant code snippet.

File: docs/patterns/existing-eks-observability-accelerators/existing-eks-awsnative-observability.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ This example makes use of CloudWatch, as a metric and log aggregation layer. In
 
 ## Objective
 
-This pattern aims to add Observability on top of an existing EKS cluster, with AWS services.
+This pattern aims to add Observability and control plane logging on top of an existing EKS cluster, with AWS services.
```
Comment (elamaran11) [inline docs/patterns/existing-eks-observability-accelerators/existing-eks-awsnative-observability.md:17]: This change is not required. I would rather add a bullet to say. If Control Plane logging is not enabled.. please use enableControlPlane logging function attaching a small relevant code snippet.

File: docs/patterns/existing-eks-observability-accelerators/existing-eks-mixed-observability.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ Utilizing CloudWatch and X-Ray as an aggregation layer allows for a fully-manage
 
 ## Objective
 
-This pattern aims to add Observability on top of an existing EKS cluster, with a mixture of AWS native and open source managed AWS services.
+This pattern aims to add Observability and control plane logging on top of an existing EKS cluster, with a mixture of AWS native and open source managed AWS services.
```
Comment (elamaran11) [inline docs/patterns/existing-eks-observability-accelerators/existing-eks-mixed-observability.md:17]: This change is not required. I would rather add a bullet to say. If Control Plane logging is not enabled.. please use enableControlPlane logging function attaching a small relevant code snippet.

File: docs/patterns/existing-eks-observability-accelerators/existing-eks-nginx-observability.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ The current example deploys the AWS Distro for OpenTelemetry Operator for Amazon
 
 ## Objective 
 
-This pattern aims to add Observability on top of an existing EKS cluster and NGINX workloads, with open source managed AWS services.
+This pattern aims to add Observability and control plane logging on top of an existing EKS cluster and NGINX workloads, with open source managed AWS services.
```
Comment (elamaran11) [inline docs/patterns/existing-eks-observability-accelerators/existing-eks-nginx-observability.md:13]: This change is not required. I would rather add a bullet to say. If Control Plane logging is not enabled.. please use enableControlPlane logging function attaching a small relevant code snippet.

File: docs/patterns/existing-eks-observability-accelerators/existing-eks-opensource-observability.md [hunk 5]
```diff
@@ -12,6 +12,7 @@ please read the [service documentation](https://docs.aws.amazon.com/eks/latest/u
 ## Objective
 
 Configure the existing Amazon EKS cluster with below Observability components;
+- Control Plane logging
```
Comment (elamaran11) [inline docs/patterns/existing-eks-observability-accelerators/existing-eks-opensource-observability.md:15]: This change is not required. I would rather add a bullet to say. If Control Plane logging is not enabled.. please use enableControlPlane logging function attaching a small relevant code snippet.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-apiserver-opensource-observability.md [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 ## Objective
 
-This pattern demonstrates how to use the _New EKS Cluster Open Source Observability Accelerator_ with API Server monitoring.
+This pattern demonstrates how to use the _New EKS Cluster Open Source Observability Accelerator_ with API Server monitoring and control plane logging.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-apiserver-opensource-observability.md:5]: Remove, need not required.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-awsnative-fargate-observability.md [hunk 7]
```diff
@@ -20,7 +20,7 @@ please read the [service documentation](https://docs.aws.amazon.com/eks/latest/u
 
 ## Objective
 
-- Deploys one production grade Amazon EKS Fargate cluster.
+- Deploys one production grade Amazon EKS Fargate cluster with control plane logging.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-awsnative-fargate-observability.md:23]: Control Plane logging can be sep bullet.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-awsnative-observability.md [hunk 8]
```diff
@@ -17,7 +17,7 @@ please read the [service documentation](https://docs.aws.amazon.com/eks/latest/u
 
 ## Objective
 
-- Deploys one production grade Amazon EKS cluster.
+- Deploys one production grade Amazon EKS cluster with control plane logging.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-awsnative-observability.md:20]: Control Plane logging can be sep bullet.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-cluster.md [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 ## Objective
 
-This pattern deploys one production grade Amazon EKS cluster, without any Observability add-on.
+This pattern deploys one production grade Amazon EKS cluster with control plane logging, without any Observability add-on.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-cluster.md:5]: remove

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md [hunk 10]
```diff
@@ -12,7 +12,7 @@ Amazon Managed Grafana is a managed service for Grafana, a popular open-source a
 
 ## Objective
 
-This pattern deploys an Amazon EKS cluster with a node group that includes instance types featuring NVIDIA GPUs.
+This pattern deploys an Amazon EKS cluster with control plane logging and a node group that includes instance types featuring NVIDIA GPUs.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:15]: Add to end of objective

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-graviton-opensource-observability.md [hunk 11]
```diff
@@ -20,7 +20,7 @@ Visit our [EKS Blueprints docs](https://github.com/aws-quickstart/cdk-eks-bluepr
 
 ## Objective
 
-- Deploys one production grade Amazon EKS cluster running on a Graviton3 Processor
+- Deploys one production grade Amazon EKS cluster running on a Graviton3 Processor with control plane logging
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-graviton-opensource-observability.md:23]: Control Plane logging can be sep bullet.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-java-opensource-observability.md [hunk 12]
```diff
@@ -2,7 +2,7 @@
 
 ## Objective
 
-This pattern demonstrates how to use the _New EKS Cluster Open Source Observability Accelerator_ with Java based workloads.
+This pattern demonstrates how to use the _New EKS Cluster Open Source Observability Accelerator_ with Java based workloads and control plane logging.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-java-opensource-observability.md:5]: Add to end of objective

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-mixed-observability.md [hunk 13]
```diff
@@ -14,7 +14,7 @@ Utilizing CloudWatch and X-Ray as an aggregation layer allows for a fully-manage
 
 ## Objective
 
-- Deploys one production grade Amazon EKS cluster.
+- Deploys one production grade Amazon EKS cluster with Control Plane logging.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-mixed-observability.md:17]: Separate item

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-nginx-opensource-observability.md [hunk 14]
```diff
@@ -2,7 +2,7 @@
 
 ## Objective
 
-This pattern demonstrates how to use the _New EKS Cluster Open Source Observability Accelerator_ with Nginx based workloads.
+This pattern demonstrates how to use the _New EKS Cluster Open Source Observability Accelerator_ with Nginx based workloads and control plane logging.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-nginx-opensource-observability.md:5]: Add to end of objective

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-opensource-observability.md [hunk 15]
```diff
@@ -14,7 +14,7 @@ please read the [service documentation](https://docs.aws.amazon.com/eks/latest/u
 
 ## Objective
 
-- Deploys one production grade Amazon EKS cluster.
+- Deploys one production grade Amazon EKS cluster with Control Plane logging.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-opensource-observability.md:17]: Separate bullet

File: lib/existing-eks-opensource-observability-pattern/index.ts [hunk 16]
```diff
@@ -120,6 +120,7 @@ export default class ExistingEksOpenSourceobservabilityPattern {
             .version('auto')
             .withAmpProps(ampAddOnProps)
             .enableOpenSourcePatternAddOns()
+            .enableControlPlaneLogging()
```
Comment (elamaran11) [inline lib/existing-eks-opensource-observability-pattern/index.ts:123]: Remove this has to be optional if not enabled already

File: lib/existing-eks-mixed-observability-pattern/index.ts [hunk 17]
```diff
@@ -55,6 +55,7 @@ export default class ExistingEksMixedobservabilityPattern {
             .region(region)
             .version('auto')
             .enableMixedPatternAddOns()
+            .enableControlPlaneLogging()
```
Comment (elamaran11) [inline lib/existing-eks-mixed-observability-pattern/index.ts:58]: Remove this has to be optional if not enabled already

File: lib/existing-eks-awsnative-observability-pattern/index.ts [hunk 18]
```diff
@@ -45,6 +45,7 @@ export default class ExistingEksAwsNativeObservabilityPattern {
             .region(region)
             .version('auto')
             .enableNativePatternAddOns()
+            .enableControlPlaneLogging()
```
Comment (elamaran11) [inline lib/existing-eks-awsnative-observability-pattern/index.ts:48]: Remove this has to be optional if not enabled already

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@5herlocked Nice work on this PR and blueprints PR. Have some tactical and doc comments. Please fix these when you get chance. We can merge this only after blueprints release so we should also factor for changes to `package.json` later.

File: docs/patterns/existing-eks-observability-accelerators/existing-eks-apiserver-observability.md [hunk 19]
```diff
@@ -20,6 +20,16 @@ You will also need:
 
 ## Deploying
 
+!!! note If control plane logging is not enabled in the existing cluster, edit `lib/existing-eks-opensource-observability-pattern/index.ts` to include the following code snippet:
```
Comment (elamaran11) [inline docs/patterns/existing-eks-observability-accelerators/existing-eks-apiserver-observability.md:23]: Can we say `to include .enableControlPlaneLogging()  as show below`.

File: docs/patterns/existing-eks-observability-accelerators/existing-eks-awsnative-observability.md [hunk 20]
```diff
@@ -31,6 +31,15 @@ You will also need:
 2. An OpenID Connect (OIDC) provider, associated to the above EKS cluster (Note: Single EKS Cluster Pattern takes care of that for you)
 
 ## Deploying
+!!! note If control plane logging is not enabled in the existing cluster, edit `lib/existing-eks-awsnative-observability-pattern/index.ts` to include the following code snippet:
```
Comment (elamaran11) [inline docs/patterns/existing-eks-observability-accelerators/existing-eks-awsnative-observability.md:34]: Can we say `to include .enableControlPlaneLogging()  as show below`.

File: docs/patterns/existing-eks-observability-accelerators/existing-eks-mixed-observability.md [hunk 21]
```diff
@@ -32,6 +32,16 @@ You will also need:
 
 ## Deploying
 
+!!! note If control plane logging is not enabled in the existing cluster, edit `lib/existing-eks-mixed-observability-pattern/index.ts` to include the following code snippet:
+```typescript
```
Comment (elamaran11) [inline docs/patterns/existing-eks-observability-accelerators/existing-eks-mixed-observability.md:36]: Can we say `to include .enableControlPlaneLogging()  as show below`.

File: docs/patterns/existing-eks-observability-accelerators/existing-eks-nginx-observability.md [hunk 22]
```diff
@@ -28,6 +28,15 @@ You will also need:
 
 ## Deploying
 
+!!! note If control plane logging is not enabled in the existing cluster, edit `lib/existing-eks-opensource-observability-pattern/index.ts` to include the following code snippet:
```
Comment (elamaran11) [inline docs/patterns/existing-eks-observability-accelerators/existing-eks-nginx-observability.md:31]: Can we say `to include .enableControlPlaneLogging()  as show below`.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@5herlocked Looks great, fix these minor issues. We will block the e2e and merge until we have blueprints release.

**Comment by 5herlocked** [comment]
Resolved the comments in the docs, ready to GTM after deployment of the upstream changes.

**Comment by elamaran11** [comment]
@5herlocked GH Failure needs to be fixed.

**Comment by 5herlocked** [comment]
Error has nothing to do with my code.

**Comment by elamaran11** [comment]
> Error has nothing to do with my code.

This is not because of your code but did you pull from main, that could be the reason. I had one PR yesterday and we never had that error.

**Review by elamaran11 — APPROVED** [review]
LGTM

**Pull request merged**